deployment/clusters.tf

```diff
@@ -36,6 +52,27 @@ resource "google_container_node_pool" "primary" {
   }
 }
 
+resource "google_container_node_pool" "benchmark" {
```

Comment:
Can you break the node pools down into modules? E.g., like was done here https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/deployment/modules/composer_env/main.tf for google_composer_environment

You can create another directory under https://github.com/GoogleCloudPlatform/ml-auto-solutions/tree/master/deployment/modules, probably something like `node_pool`, and add a `main.tf` and `variables.tf` file, and create a module with those?

main.tf could probably be something like:

```
resource "google_container_node_pool" "node_pool" {
  name = var.node_pool_name
  project = var.node_pool_project
  location = var.node_pool_location
  ...etc
}
```

Then replace what you have here with

```
module "node_pool" {
  node_pool_name = "benchmark-pool"
  etc
}
```

Breaking down the terraform configs like this makes them more modular and easy to use.

Reply:
Check if we want to keep `google_container_cluster` still inside cluster.tf but move all `google_container_node_pool` under deployment/modules/nodepool/...?

Reply:
yeah that would be good tbh, since you're adding another cluster. You can do `google_container_cluster` and `google_container_node_pool` in separate modules, I think putting them together would get a bit complicated.

Reply:
Sounds good to me, thanks!

Reply:
Do you know why `terraform plan` can't detect my changes after moving pool related to the new folder `deployment/modules/nodepool`? It only reported node pool got destroyed because I removed node pool from clusters.tf file into the new location.

Reply:
I moved cluster under deployment/modules/cluster_node_pool/ also since the cluster and pool share the same variables like `cluster, location, project` etc..

Reply:
can you share the results of `terraform plan`?